coins/bitcoin/src/rpc_helper.rs

```diff
@@ -0,0 +1,316 @@
+use std::{
+    error,
+    fmt::{self, Display},
+    io, result,
+};
+use thiserror::Error;
+use bitcoin::{
+    self,
+    hashes::{
+        hex::{self, FromHex, ToHex},
+    },
+    secp256k1, OutPoint, Transaction,
+};
+use serde::{de::Error as SerdeError, ser, Deserialize, Serialize};
+use bitcoincore_rpc_json::{serde_hex, GetRawTransactionResult};
+
+#[derive(Clone, PartialEq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetBlockWithDetailResult {
+    pub hash: bitcoin::BlockHash,
+    pub confirmations: i32,
+    pub size: usize,
+    pub strippedsize: Option<usize>,
+    pub weight: usize,
+    pub height: usize,
+    pub version: i32,
+    #[serde(default, with = "serde_hex::opt")]
+    pub version_hex: Option<Vec<u8>>,
+    pub merkleroot: bitcoin::TxMerkleNode,
+    pub tx: Vec<GetRawTransactionResult>,
+    pub time: usize,
+    pub mediantime: Option<usize>,
+    pub nonce: u32,
+    pub bits: String,
+    pub difficulty: f64,
+    #[serde(with = "serde_hex")]
+    pub chainwork: Vec<u8>,
+    pub n_tx: usize,
+    pub previousblockhash: Option<bitcoin::BlockHash>,
+    pub nextblockhash: Option<bitcoin::BlockHash>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct RpcResponseError {
+    pub code: i64,
+    pub message: String,
+}
+
+impl Default for RpcResponseError {
+    fn default() -> Self {
+        RpcResponseError {
+            code: -1,
+            message: String::from(""),
+        }
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub struct RpcResponse<T> {
```

Comment:
I believe if this is written as an enum, we can ensure it's either result or error, not both, resolving a prior comment.

Should support Clone.